src/mgmt/cli_mgmt.c

```diff
@@ -367,6 +369,28 @@ get_attach_info(const char *name, int *npsrs, struct dc_mgmt_psr **psrs)
 	*npsrs = resp->n_psrs;
 	*psrs = p;
 
+	if (sy_info) {
+		size = sizeof(sy_info->provider);
+		strncpy(sy_info->provider, resp->provider, size);
+		sy_info->provider[size-1] = '\0';
```

Comment:
I will admit that my C is a little rusty, but this seems strange to me. I guess the idea is that if `resp->provider` is longer than `sy_info->provider`, this will copy as much as it can and then ensure that the resulting string is null-terminated?

Wouldn't it be better to check the length of `resp->provider` and fail with DER_INVAL if it would be truncated?

Reply:
Your understanding is correct.  And, you also make a good point that it would be better to fail the operation instead.  I was focused on keeping it safe, but we can have 'safe' and 'better' at the same time by failing it if the source string is too long.  I'll make the change.